@aloubyansky can you take a look at the failure:
```
[ERROR] testCustomPackaging  Time elapsed: 3.598 s  <<< FAILURE!
org.opentest4j.AssertionFailedError: 

Expecting:
 <1>
to be equal to:
 <0>
but was not.
	at io.quarkus.maven.it.PackageIT.testCustomPackaging(PackageIT.java:67)
```

--- a/devtools/maven/src/main/java/io/quarkus/maven/BuildMojo.java
+++ b/devtools/maven/src/main/java/io/quarkus/maven/BuildMojo.java
@@ -165,7 +165,8 @@ public class BuildMojo extends AbstractMojo {
     public void execute() throws MojoExecutionException {
         final Artifact projectArtifact = project.getArtifact();
         final AppArtifact appArtifact = new AppArtifact(projectArtifact.getGroupId(), projectArtifact.getArtifactId(),
-                projectArtifact.getClassifier(), projectArtifact.getType(), projectArtifact.getVersion());
+                projectArtifact.getClassifier(), projectArtifact.getArtifactHandler().getExtension(),
+                projectArtifact.getVersion());
         final AppModel appModel;
         final BootstrapAppModelResolver modelResolver;
         try {

--- a/devtools/maven/src/test/java/io/quarkus/maven/it/PackageIT.java
+++ b/devtools/maven/src/test/java/io/quarkus/maven/it/PackageIT.java
@@ -1,13 +1,17 @@
 package io.quarkus.maven.it;
 
 import static org.assertj.core.api.Assertions.assertThat;
+import static org.junit.Assert.assertEquals;
 
 import java.io.File;
 import java.io.FileNotFoundException;
 import java.util.Collections;
 import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Set;
 
 import org.apache.maven.shared.invoker.MavenInvocationException;
+import org.assertj.core.util.Arrays;
 import org.junit.jupiter.api.Test;
 
 import io.quarkus.maven.it.verifier.MavenProcessInvocationResult;
@@ -50,6 +54,33 @@ public class PackageIT extends MojoTestBase {
         final File targetDir = getTargetDir();
         assertThat(getNumberOfFilesEndingWith(targetDir, ".jar")).isEqualTo(1);
         assertThat(getNumberOfFilesEndingWith(targetDir, ".original")).isEqualTo(1);
+    }
+
+    @Test
+    public void testCustomPackaging()
+            throws MavenInvocationException, FileNotFoundException, InterruptedException {
+        testDir = initProject("projects/custom-packaging-plugin", "projects/custom-packaging-plugin");
+
+        running = new RunningInvoker(testDir, false);
+        MavenProcessInvocationResult result = running.execute(Collections.singletonList("install"),
+                Collections.emptyMap());
+        assertThat(result.getProcess().waitFor()).isEqualTo(0);
+
+        testDir = initProject("projects/custom-packaging-app", "projects/custom-packaging-app");
+
+        running = new RunningInvoker(testDir, false);
+        result = running.execute(Collections.singletonList("package"),
+                Collections.emptyMap());
+        assertThat(result.getProcess().waitFor()).isEqualTo(0);
+
+        final File targetDir = getTargetDir();
+        final File[] files = targetDir.listFiles(f -> f.getName().endsWith(".jar"));
+        Set<String> jarNames = new HashSet<>(files.length);
+        for (File f : files) {
+            jarNames.add(f.getName());
+        }
+        assertEquals(new HashSet<>(Arrays.asList(new String[] { "acme-custom-packaging-app-1.0-SNAPSHOT-runner.jar",
+                "acme-custom-packaging-app-1.0-SNAPSHOT.jar" })), jarNames);
     }
 
     private int getNumberOfFilesEndingWith(File dir, String suffix) {

--- a/devtools/maven/src/test/resources/projects/custom-packaging-app/pom.xml
+++ b/devtools/maven/src/test/resources/projects/custom-packaging-app/pom.xml
@@ -1,0 +1,46 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/maven-v4_0_0.xsd">
+
+  <modelVersion>4.0.0</modelVersion>
+
+  <groupId>org.acme</groupId>
+  <artifactId>acme-custom-packaging-app</artifactId>
+  <packaging>blah</packaging>
+  <version>1.0-SNAPSHOT</version>
+
+  <dependencyManagement>
+    <dependencies>
+      <dependency>
+        <groupId>io.quarkus</groupId>
+        <artifactId>quarkus-bom</artifactId>
+        <version>@project.version@</version>
+        <type>pom</type>
+        <scope>import</scope>
+      </dependency>
+    </dependencies>
+  </dependencyManagement>
+
+  <build>
+    <plugins>
+      <plugin>
+        <groupId>io.quarkus</groupId>
+        <artifactId>quarkus-maven-plugin</artifactId>
+         <version>@project.version@</version>
+        <executions>
+          <execution>
+            <goals>
+              <goal>build</goal>
+            </goals>
+          </execution>
+        </executions>
+      </plugin>
+      <plugin>
+        <groupId>org.acme</groupId>
+        <artifactId>acme-custom-packaging-plugin</artifactId>
+        <version>1.0-SNAPSHOT</version>
+        <extensions>true</extensions>
+      </plugin>
+    </plugins>
+  </build>
+</project>

--- a/devtools/maven/src/test/resources/projects/custom-packaging-app/src/main/java/org/acme/AcmeApp.java
+++ b/devtools/maven/src/test/resources/projects/custom-packaging-app/src/main/java/org/acme/AcmeApp.java
@@ -1,0 +1,5 @@
+package org.acme;
+
+public class AcmeApp {
+    
+}

--- a/devtools/maven/src/test/resources/projects/custom-packaging-plugin/pom.xml
+++ b/devtools/maven/src/test/resources/projects/custom-packaging-plugin/pom.xml
@@ -1,0 +1,51 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project xmlns="http://maven.apache.org/POM/4.0.0"
+  xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+  xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/maven-v4_0_0.xsd">
+
+  <modelVersion>4.0.0</modelVersion>
+
+  <groupId>org.acme</groupId>
+  <artifactId>acme-custom-packaging-plugin</artifactId>
+  <version>1.0-SNAPSHOT</version>
+  <packaging>maven-plugin</packaging>
+
+  <properties>
+    <maven.compiler.source>1.8</maven.compiler.source>
+    <maven.compiler.target>1.8</maven.compiler.target>
+  </properties>
+
+  <dependencyManagement>
+    <dependencies>
+      <dependency>
+        <groupId>io.quarkus</groupId>
+        <artifactId>quarkus-bom</artifactId>
+        <version>@project.version@</version>
+        <type>pom</type>
+        <scope>import</scope>
+      </dependency>
+    </dependencies>
+  </dependencyManagement>
+
+  <build>
+    <plugins>
+      <plugin>
+        <artifactId>maven-plugin-plugin</artifactId>
+        <version>3.5.2</version>
+      </plugin>
+    </plugins>
+  </build>
+
+  <dependencies>
+    <dependency>
+      <groupId>org.apache.maven</groupId>
+      <artifactId>maven-plugin-api</artifactId>
+    </dependency>
+    <dependency>
+      <groupId>org.apache.maven.plugin-tools</groupId>
+      <artifactId>maven-plugin-annotations</artifactId>
+      <scope>provided</scope>
+    </dependency>
+  </dependencies>
+
+</project>

--- a/devtools/maven/src/test/resources/projects/custom-packaging-plugin/src/main/java/org/acme/AcmeMojo.java
+++ b/devtools/maven/src/test/resources/projects/custom-packaging-plugin/src/main/java/org/acme/AcmeMojo.java
@@ -1,0 +1,14 @@
+package org.acme;
+
+import org.apache.maven.plugins.annotations.*;
+
+@Mojo(name = "hello",
+        requiresDependencyResolution = ResolutionScope.COMPILE_PLUS_RUNTIME,
+        requiresProject = true,
+        defaultPhase = LifecyclePhase.COMPILE)
+public class AcmeMojo extends org.apache.maven.plugin.AbstractMojo {
+
+    @Override
+    public void execute() {
+    }
+}

--- a/devtools/maven/src/test/resources/projects/custom-packaging-plugin/src/main/resources/META-INF/plexus/components.xml
+++ b/devtools/maven/src/test/resources/projects/custom-packaging-plugin/src/main/resources/META-INF/plexus/components.xml
@@ -1,0 +1,34 @@
+<component-set>
+    <components>
+        <component>
+            <role>org.apache.maven.artifact.handler.ArtifactHandler</role>
+            <role-hint>blah</role-hint>
+            <implementation>org.apache.maven.artifact.handler.DefaultArtifactHandler</implementation>
+            <configuration>
+                <type>blah</type>
+                <extension>jar</extension>
+                <packaging>jar</packaging>
+                <language>java</language>
+            </configuration>
+        </component>
+        <component>
+            <role>org.apache.maven.lifecycle.mapping.LifecycleMapping</role>
+            <role-hint>blah</role-hint>
+            <implementation>org.apache.maven.lifecycle.mapping.DefaultLifecycleMapping</implementation>
+            <configuration>
+                <phases>
+                    <process-resources>org.apache.maven.plugins:maven-resources-plugin:resources</process-resources>
+                    <compile>org.acme:acme-custom-packaging-plugin:hello,org.apache.maven.plugins:maven-compiler-plugin:compile</compile>
+                    <process-test-resources>org.apache.maven.plugins:maven-resources-plugin:testResources</process-test-resources>
+                    <test-compile>org.apache.maven.plugins:maven-compiler-plugin:testCompile</test-compile>
+                    <test>org.apache.maven.plugins:maven-surefire-plugin:test</test>
+                    <package>org.apache.maven.plugins:maven-jar-plugin:jar</package>
+                    <install>org.apache.maven.plugins:maven-install-plugin:install</install>
+                    <deploy>org.apache.maven.plugins:maven-deploy-plugin:deploy</deploy>
+                </phases>
+            </configuration>
+
+        </component>
+
+    </components>
+</component-set>

